app/fragments/secure/components/TransferSingle.tsx

```diff
@@ -127,9 +129,10 @@ export const TransferSingle = memo((props: ConfirmLoadedPropsSingle) => {
         target = jettonTarget;
     }
 
-    const friendlyTarget = target.address.toString({ testOnly: isTestnet, bounceable: target.bounceable });
-    // Contact wallets
-    const contact = useContact(friendlyTarget);
+    const friendlyTarget = target.address.toString({ testOnly: isTestnet });
+    const friendlyTargetNonBounceable = target.address.toString({ testOnly: isTestnet, bounceable: false });
+    // Previously contacts could be created with different address formats, now it's only bounceable, but we need to check both formats to keep compatibility
+    const contact = addressBook.asContact(friendlyTarget) || addressBook.asContact(friendlyTargetNonBounceable);
```

Comment:
You could incapsulate this logic inside of asContact method (parsing Address and checking for both formats)